engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/ModelData.scala

```diff
@@ -84,9 +93,21 @@ trait ModelData extends AutoCloseable {
 
   def modelClassLoader : ModelClassLoader
 
-  def processConfigFromConfiguration: ModelConfigToLoad
+  protected def inputConfig: Config
+
+  protected def modelConfigLoader: ModelConfigLoader
+
+  lazy val processConfig: Config = modelConfigLoader.resolveConfigDuringExecution(configToPassInExecution, modelClassLoader.classLoader)
+
+  // Config to pass in execution (see FlinkProcessManager).
+  private lazy val configToPassInExecution: Config = {
+    val defaultConfigDuringExecution = modelConfigLoader.resolveConfigDuringExecution(inputConfig, modelClassLoader.classLoader)
```

Comment:
I'm a little bit supprised that we here firstly invoke `resolveConfigDuringExecution` and then `resolveConfigToPassInExecution`. because we are not yet "during execution". And am I right that during execution config passed in execution is resolved one more time using `resolveConfigDuringExecution`?

Reply:
`resolveConfigDuringExecution` is invoked twice because:
1. `defaultConfigDuringExecution` (I will rename it to `configDuringModelAnalysis`) is `processTypes.{type_name}.modelConfig` + `defaultModelConfig.conf` + `reference.conf`.
2. In order to modify config send to Flink process (`processTypes.{type_name}.modelConfig`) we would like to have access to all the config above. Some urls/paths/etc. can be defined in resources that are possibly needed to modify config.

Reply:
Why not to: 
1. invoke here only `resolveConfigToPassInExecution` which by default will resolve this by merging `processTypes.{type_name}.modelConfig + defaultModelConfig.conf + reference.conf`
2. after that do step 3.
Result will be the same but it will be more straightforward: only two steps,  neither of them done more then once

Reply:
Currently all the methods are called only once. Since `resolveConfigUsingDefaults` is extracted I think it makes it more clear.